src/app/features/profile/profile-detail/profile-detail.component.ts

```diff
@@ -1,23 +1,18 @@
-import { Component, OnInit } from '@angular/core';
-import { Store } from '@ngrx/store';
-import { profileActions } from '@store/actions';
-import { AppState } from '@store/reducers';
-import { getUserProfile } from '@store/selectors';
+import { Component, Input } from '@angular/core';
+import { UserProfile } from '@interfaces';
 
 @Component({
     selector: 'app-profile-detail',
     styleUrls: ['./profile-detail.component.less'],
     templateUrl: './profile-detail.component.html'
 })
-export class ProfileDetailComponent implements OnInit {
+export class ProfileDetailComponent {
 
-    user$ = this.store.select(getUserProfile);
+    @Input() user: UserProfile;
 
-    constructor (private store: Store<AppState>) {}
+    public goBack () {
 
-    ngOnInit () {
-
-        this.store.dispatch(profileActions.initProfile());
+        this.user.displayed = false;
```

Comment:
this is bad practice, provided through the Input values should be immutable
to output something you need to create use Output or ngrx actions (if global)